File: r-node-diamonds/r-scripts/scripts/runner.R
library("tidyverse")
library("randomForest")
library("jsonlite")

args = commandArgs(trailingOnly=TRUE)
if (length(args) < 1) {
  stop("The parameter -> diamond json to be valued.", call.=FALSE)
} 

cuts = c(1,2,3,4,5)
names(cuts) = c("Fair", "Good", "Very Good", "Premium", "Ideal")

colors = c(1,2,3,4,5,6,7)
names(colors) = c("D", "E", "F", "G", "H", "I", "J")

clarities = c(1,2,3,4,5,6,7,8)
names(clarities) = c("I1", "SI2", "SI1", "VS2", "VS1", "VVS2", "VVS1", "IF")

diamond_json = args[1]
diamond <- fromJSON(diamond_json) 
diamond$cut <- unname(cuts[diamond$cut])
diamond$color <-unname(colors[diamond$color])
diamond$clarity <- unname(clarities[diamond$clarity])

model <- readRDS("/Users/jwszol/Documents/workspace-git/ml-kraken-models/r-node-diamonds/r-scripts/models/randomForestThin.rds")
final_predictions <- predict(model, diamond) %>% toJSON() %>% cat()

File: python-node-diamonds/py-scripts/scripts/runner.py
import argparse
import pandas as pd
import numpy as np
import joblib
import json
from sklearn import preprocessing
from sklearn.utils import column_or_1d

parser = argparse.ArgumentParser("pservice")
parser.add_argument("data", help="Json string with diamond(s) features to compute price", type=str)
args = parser.parse_args()

class MyLabelEncoder(preprocessing.LabelEncoder):
    def fit(self, y):
        y = column_or_1d(y, warn=True)
        self.classes_ = pd.Series(y).unique()
        return self

cuts = ['Fair', 'Good', 'Ideal', 'Premium', 'Very Good']
colors = ['D', 'E', 'F', 'G', 'H', 'I', 'J']
clarities = ['I1', 'SI2', 'SI1', 'VS2', 'VS1', 'VVS2', 'VVS1', 'IF']

data = pd.read_json(args.data)

to_encode_column = ['cut', 'color', 'clarity']
to_encode_names = [cuts, colors, clarities]

for idx, c in enumerate(to_encode_column):
    le = MyLabelEncoder()
    le.fit(to_encode_names[idx])
    data[c] = le.transform(data[c])

data = data.drop(columns=['x', 'z', 'table'])

model = joblib.load("/app/py-scripts/models/randomForest.sav")

results = model.predict(data)

print(json.dumps(np.ndarray.tolist(results)))
